app/models/miq_expression.rb

```diff
@@ -575,7 +575,7 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sql = Expression.build(@pexp, tz).to_sql if @pexp.present?
```

Comment:
> Note, naming is hard. We have @pexp, @exp and the resulting objects of .build which are also Expression-like... Can we come up with better names? I don't remember this code so I have really no idea how they're all different.

Yep, this is the main reason this change felt partly :cry: for me. I toyed with the idea of calling them Operators (which is what they sort of represent), but felt wrong to lump that in with their operands (the field/tag objects that we're injecting, and the values which have state-dependent things going on). I'll try to come up with something better.....@gtanzillo any opinions here?

Renaming `@pexp` seems like a fairly small change that I can incorporate here :thumbsup: (for the record it's the form of the `@exp` hash that has had things that can't be evaluated in SQL stripped out). I want that to go away eventually, but wishful thinking for now.

`@exp` will be harder. There are tons of references to it and I think it's not encapsulated by the accessor (loading from YAML won't use it).